unifiedpush_android/lib/unifiedpush_android.dart

```diff
@@ -0,0 +1,133 @@
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:unifiedpush_platform_interface/unifiedpush_platform_interface.dart';
+
+import 'CallbackDispatcher.dart';
+import 'constants.dart';
+
+
+class UnifiedPushAndroid extends UnifiedPushPlatform {
+  static void registerWith() {
+    UnifiedPushPlatform.instance = UnifiedPushAndroid();
+  }
+  
+  static SharedPreferences? _prefs;
+  static get prefs async {
+    if (_prefs == null) {
+      _prefs = await SharedPreferences.getInstance();
+    }
+    return _prefs;
+  }
```

Comment:
Since the return type isn't specified, it's dynamic by default. This causes it to return a `Future<dynamic>` on the first run, which causes the following error.
```
E/flutter (27055): Receiver: Instance of 'Future<dynamic>'
E/flutter (27055): Tried calling: setInt("unifiedpush/method:onNewEnpoint", 8284117098658340351)
E/flutter (27055): #0      Object.noSuchMethod (dart:core-patch/object_patch.dart:68:5)
E/flutter (27055): #1      UnifiedPushAndroid.initializeBackgroundCallback (package:unifiedpush_android/unifiedpush_android.dart:75:13)
E/flutter (27055): #2      UnifiedPush.initializeWithCallbackInstantiated (package:unifiedpush/unifiedpush.dart:83:40)
E/flutter (27055): <asynchronous suspension>
E/flutter (27055): #3      UnifiedPush.initializeWithCallback (package:unifiedpush/unifiedpush.dart:53:5)
E/flutter (27055): <asynchronous suspension>
```
Line 75 of this file expects `SharedPreferences` not a `Future`.

The solution would probably be to define the return type, and then just await `prefs` on every call.
```suggestion
  static SharedPreferences? _prefs;
  static Future<SharedPreferences> get prefs async {
    if (_prefs == null) {
      _prefs = await SharedPreferences.getInstance();
    }
    return _prefs!;
  }
```

Reply:
Ah yeah me stupid, I fixed that in the `unifiedpush` module but not in the Android impl.

Reply:
I changed from a getter to a method, having an async getter is... weird on usage :)